src/app/scripts/cac/map/cac-map-control.js

```diff
@@ -0,0 +1,127 @@
+    var overlays = {};
+    var stamenTonerAttribution = [
+        'Map tiles by <a href="http://stamen.com">Stamen Design</a>, ',
+        'under <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a>. ',
+        'Data by <a href="http://openstreetmap.org">OpenStreetMap</a>, ',
+        'under <a href="http://www.openstreetmap.org/copyright">ODbL</a>.'
+    ].join('');
+    var stamenAttribution = [
+        'Map tiles by <a href="http://stamen.com">Stamen Design</a>, ',
+        'under <a href="http://creativecommons.org/licenses/by/3.0">CC BY 3.0</a>. ',
+        'Data by <a href="http://openstreetmap.org">OpenStreetMap</a>, ',
+        'under <a href="http://creativecommons.org/licenses/by-sa/3.0">CC BY SA</a>.'
+    ].join('');
+
+    function MapControl(options) {
+        this.options = $.extend({}, defaults, options);
+        overlaysControl = new CAC.Map.OverlaysControl();
+        map = L.map(this.options.id).setView(this.options.center, this.options.zoom);
+
+
+        initializeBasemaps();
+        initializeOverlays();
+        initializeLayerControl();
+    }
+
+    MapControl.prototype.locateUser = locateUser;
+    MapControl.prototype.plotLocations = plotLocations;
+
+    return MapControl;
+
+    function initializeBasemaps() {
+        basemaps.Streets = L.tileLayer('https://stamen-tiles-{s}.a.ssl.fastly.net/toner-lite/{z}/{x}/{y}.png', {
+            attribution: stamenTonerAttribution
+        });
+
+        basemaps.Terrain = L.tileLayer('https://stamen-tiles-{s}.a.ssl.fastly.net/terrain/{z}/{x}/{y}.png', {
+            attribution: stamenAttribution
+        });
+
+        basemaps.Streets.addTo(map);
+    }
+
+    function initializeOverlays() {
+        overlays['Bike Share Locations'] = overlaysControl.bikeShareOverlay();
+        overlays['Bike Parking'] = overlaysControl.bikeParkingOverlay();
+    }
+
+    function initializeLayerControl() {
+        L.control.layers(basemaps, overlays, {
+            position: 'bottomright'
+        }).addTo(map);
+    }
+
+    /**
+     * Use HTML5 navigator to locate user and place a circle at their estimated location.
+     *  Modify the target variable (instantiated above) to enable the killing of this watch job
+     *  once a target location has been reached
+     *
+     * @return {object} (promise) which should resolve to the current coordinates of a user
+     */
+    function locateUser() {
+        var deferred = $.Deferred();
+        var options = {
+            enableHighAccuracy: true,
+            timeout: 5000,
+            maximumAge: 0
+        };
+        var success = function(pos) {
+            var latlng = [pos.coords.latitude, pos.coords.longitude];
+            if (userMarker) {
+                userMarker.setLatLng(latlng);
+            } else {
+                userMarker = new L.CircleMarker(latlng)
+                  .on('click', function() {
+                      $(document).trigger('MOS.Map.Control.CurrentLocationClicked', latlng);
```

Comment:
Typically events are fired either by wrapping the object (in this case MapControl) itself, or exposing an events property that you fire the events on. This avoids polluting the global namespace and having a bunch of other event checkers execute since the document is the tree root for event propagation.

So, an example:
```
// In Map.Control
function MapControl(options) {
    ...
   this.events = $({});
}

function locateUser() {
    ...
    MapControl.events.trigger('MOS.Map.Control.CurrentLocationClicked', latlng);
}

// In Map.Pages
...
mapControl.events.on('MOS.Map.Control.CurrentLocationClicked', function () { ... });
```
}